nekoyume/broadcast.py

```diff
@@ -28,105 +28,130 @@
-            res = broadcast(url, serialized, node, my_node)
-            if res.status_code == 403:
-                result = res.json()
-                # 0 is Genesis block.
-                block_id = result.get('block_id', 0)
-                query = db.session.query(Block).filter(
-                    Block.id.between(block_id, serialized['id'])
-                ).order_by(Block.id)
-                offset = 0
-                while True:
-                    sync_blocks = query[
-                        offset:offset+BROADCAST_LIMIT
-                    ]
-                    # TODO bulk api
-                    for block in sync_blocks:
-                        s = block.serialize(
-                            use_bencode=False,
-                            include_suffix=True,
-                            include_moves=True,
-                            include_hash=True
-                        )
-                        broadcast(url, s)
-                    offset += BROADCAST_LIMIT
-                    if len(sync_blocks) < BROADCAST_LIMIT:
-                        break
-        except (ConnectionError, Timeout):
-            continue
-    db.session.commit()
-
-
-def broadcast_move(
-        serialized: Mapping[str, object],
-        sent_node: Optional[Node]=None,
-        my_node: Optional[Node]=None,
-) -> None:
-    """
-    It broadcast this move to every nodes you know.
-
-    :param      serialized: serialized :class:`nekoyume.move.Move`.
-                            that will be broadcasted.
-    :param       sent_node: sent :class:`nekoyume.node.Node`.
-                            this node ignore sent node.
-    :param         my_node: my :class:`nekoyume.node.Node`.
-                            received node ignore my node when they
-                            broadcast received object.
-    """
-    for node in filter_nodes(sent_node):
-        try:
-            url = urllib.parse.urljoin(node.url, POST_MOVE_ENDPOINT)
-            broadcast(url, serialized, node, my_node)
-        except (ConnectionError, Timeout):
-            continue
-
-    db.session.commit()
+class Broadcaster:
+
+    @classmethod
+    def broadcast(
+            cls,
+            serialized: Mapping[str, object],
+            sent_node: Optional[Node]=None,
+            my_node: Optional[Node]=None,
+    ):
+        for node in filter_nodes(sent_node):
+            try:
+                cls._broadcast(
+                    serialized=serialized,
+                    sent_node=node,
+                    my_node=my_node
+                )
+            except (ConnectionError, Timeout):
+                continue
+        db.session.commit()
+
+    @classmethod
+    def _broadcast(
+            cls,
+            serialized: Mapping[str, object],
+            sent_node: Optional[Node]=None,
+            my_node: Optional[Node]=None,
+    ) -> None:
+        raise NotImplementedError
```

Comment:
It's unnatural for me to define a hierarchy of classes that consist of only class methods but no instance methods, since it actually does nothing with subtype polymorphism.  These two class methods seem to purpose to make a hook while maintaining the larger pattern.  IMHO simple callbacks are way easier to understand, e.g.:

```python
def multicast(
    serialized: Mapping[str, object],
    notify: Callable[
        [Mapping[str, object], Optional[Node], Optional[Node]],
        None
    ],
    sent_node: Optional[Node]=None,
    my_node: Optional[Node]=None,
) -> None:
    for node in filter_nodes(sent_node):
        try:
            notify(
                serialized=serialized,
                sent_node=node,
                my_node=my_node
            )
        except (ConnectionError, Timeout):
            continue
    db.session.commit()

def notify_block(
    serialized: Mapping[str, object],
    sent_node: Optional[Node]=None,
    my_node: Optional[Node]=None,
) -> None:
    ...

def notify_move(
    serialized: Mapping[str, object],
    sent_node: Optional[Node]=None,
    my_node: Optional[Node]=None,
) -> None:
    ...

def notify_node(
    serialized: Mapping[str, object],
    sent_node: Optional[Node]=None,
    my_node: Optional[Node]=None,
) -> None:
    ...
```